accounts/views.py

```diff
@@ -36,8 +33,7 @@ def get(self, request):
         user = auth.authenticate(remote_user=pennkey, shibboleth_attributes=shibboleth_attributes)
         if user:
             auth.login(request, user)
-            params = request.get_full_path().split('next=')[1]
-            return redirect('https://platform.pennlabs.org' + params)
+            return redirect(request.GET.get('next'))
```

Comment:
Does `GET['next']` contain the redirect URL for success? Is `pennlabs.org` hardcoded anywhere? Just curious about design choices.

Reply:
Yep, that's it. It's a relative URL to redirect once logged in. There's no reference to `pennlabs.org`. In the normal login process next should be `accounts/authorize?x=...`, Since users will be logging in through OAuth2

Reply:
Gotcha. If we don't hardcode `pennlabs.org` is there any way somebody could spoof the redirect URL back to ours with their own hosted instance of Shibboleth?

Reply:
Nope, the redirect doesn't contain any user information. At that point the user is already logged into platform and a cookie is set. The redirect is just to let the user continue the OAuth2 flow. There shouldn't be any way to leak data through this.